design-reference/architecture/VersionAndGo/readme.md

```diff
@@ -0,0 +1,113 @@
+
+## Definitions
+ - `Engineering Team (Eng Team)` - DevOps or Operations team consisting of Infrastructure Administrators or similar roles.
+ - `Application Team (App Team)` - Software Developers with an application to deploy to Azure.
+ - `Development Repo (Dev Repo)` - Cobalt's repo hosted on Github at [https://github.com/microsoft/cobal](https://github.com/microsoft/cobal)
+ - `Engineering Repo (Eng Repo)` - Individual template repository managed by the `Engineering Team`
+ - `Application Repo (App Repo)` - Repository that contains the application code to be deployed and managed by the `Application Team`
+ - `Version` - all versioning schemes should use [SemVer](https://semver.org/) or variant. 
+
+## Version and Go Setup Description
+Our design starts after the initial Cobalt setup, where an individual template is cloned into an organizations source control (ref: [Setup Doc](https://github.com/microsoft/cobalt/blob/master/docs/GETTING_STARTED_APP_DEV.md)). An application development team is ready to start deploying their code to infrastructure built in Azure. Typically they would make a request to the engineering team. The Eng team would provide the App team with a link to the latest infra configuration files for a specific template. The App team downloads this tar/zip file and extracts the folder/files.
+
+```
+infra/
+ |_ Application.tf
+ |_ Application.yaml
+```
+
+These files should be derived from the template folder in dev repo and cloned into the eng repo. Proposed folder path/location is:
+
+```
+infra/
+  |_template
+      |_template_name
+          |_tests
+          |_config
+          |   |_ application.tf
+          |   |_ application.yml
+          |_template files...
+```
+
+ - `Appliation.tf` is the artifact that points to the template in the eng repo. The application team would need to fill this file out with the requested configuration. Example:
+
+ ```
+ module "az-isolated-service-single-region" {
+   source = "git://dev.azure.com/project/repo?ver=1.0.0//infra/template/az-isolated-service-single-region/"
+   resource_group_location = "eastus"
+   name = "az-simple"
+   deployment_targets = [{
+     app_name = "cobalt-backend-api",
+     image_name = "msftcse/az-service-single-region",
+     image_release_tag_prefix = "release"
+   }]
+   acr_build_git_source_url = "https://github.com/erikschlegel/echo-server.git"
+}
+ ```
+
+ - `Application.yaml` - is the artifact that points to the azure-pipeline.yml file in the eng repo that contains the build/test/deploy pipeline.
+
+This folder/files should be checked into the root of the app repo. The Eng team would then configure a new pipeline in the app teams Azure Devops project to point to these files and deploy the requested infrastructure.
+
+> Note: We should strive to continue to use a single set of pipeline files for all pipelines referenced in this document.
+
+## Version and Go Eng Pipeline Description
```

Comment:
Not sure this is in scope for our milestone but I could see these be different yaml templates for the different publishing strategies. i.e., one for "publish to git", one to "publish to terraform repo", etc...